tests/pricing_models/hyperdrive/hd_test_calc_in_given_out.py

```diff
@@ -0,0 +1,770 @@
+"""
+Testing for the calc_in_given_out of the pricing models.
+"""
+
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-locals
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=duplicate-code
+
+from dataclasses import dataclass
+import unittest
+import numpy as np
+from elfpy.pricing_models.hyperdrive import HyperdrivePricingModel
+
+from elfpy.types import MarketState, Quantity, StretchedTime, TokenType
+
+
+@dataclass
+class TestCaseCalcInGivenOutSuccess:
+    """Dataclass for calc_in_given_out test cases"""
+
+    out: Quantity
+    market_state: MarketState
+    fee_percent: float
+    days_remaining: float
+    time_stretch_apy: float
+    test_id: int
+
+    __test__ = False  # pytest: don't test this class
+
+
+@dataclass
+class TestResultCalcInGivenOutSuccess:
+    """Dataclass for calc_in_given_out test results"""
+
+    without_fee_or_slippage: float
+    without_fee: float
+    hyperdrive_fee: float
+    hyperdrive_with_fee: float
+
+    __test__ = False  # pytest: don't test this class
+
+
+@dataclass
+class TestCaseCalcInGivenOutFailure:
+    """Dataclass for calc_in_given_out test cases"""
+
+    out: Quantity
+    market_state: MarketState
+    fee_percent: float
+    time_remaining: StretchedTime
+
+    __test__ = False  # pytest: don't test this class
+
+
+class TestCalcInGivenOut(unittest.TestCase):
```

Comment:
We're duplicating a lot of code between these test files. My preference is that we use common inputs and provide the different outputs as a dictionary keyed by the model's name. That way we keep everything in one file and all of the pricing models benefit when we add cases.